From 67e12898d53aef36ac0bacaf50261199913b3689 Mon Sep 17 00:00:00 2001
From: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Date: Wed, 14 Sep 2016 21:39:24 +0300
Subject: [PATCH 4/5] vsp-renderer: set bytesused and length of
 v4l2_surface_state to v4l2_buffer

Each plane bytesused and length is calculated in v4l2-renderer and saved
in v4l2_surface_state.
---
 src/vsp-renderer.c | 35 +++--------------------------------
 1 file changed, 3 insertions(+), 32 deletions(-)

diff --git a/src/vsp-renderer.c b/src/vsp-renderer.c
index 0f8bfa5..e985609 100644
--- a/src/vsp-renderer.c
+++ b/src/vsp-renderer.c
@@ -694,33 +694,6 @@ vsp_queue_buffer(int fd, int capture, struct vsp_surface_state *vs)
 	struct v4l2_buffer buf;
 	struct v4l2_plane planes[VIDEO_MAX_PLANES];
 	int i;
-	struct v4l2_format current_fmt;
-	unsigned int stride;
-
-	stride = vs->base.planes[0].stride;
-	if (!stride){
-
-		memset(&current_fmt, 0, sizeof(struct v4l2_format));
-		current_fmt.type = (capture) ? V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE : V4L2_BUF_TYPE_VIDEO_OUTPUT_MPLANE;
-
-		if (ioctl(fd, VIDIOC_G_FMT, &current_fmt) == -1) {
-			weston_log("VIDIOC_G_FMT failed to %d (%s).\n", fd, strerror(errno));
-		}
-
-		DBG("Current video format: %d, %08x(%c%c%c%c) %ux%u (stride %u) field %08u buffer size %u\n",
-		    current_fmt.type,	
-		    current_fmt.fmt.pix_mp.pixelformat,
-		    (current_fmt.fmt.pix_mp.pixelformat >> 24) & 0xff,
-		    (current_fmt.fmt.pix_mp.pixelformat >> 16) & 0xff,
-		    (current_fmt.fmt.pix_mp.pixelformat >>  8) & 0xff,
-		    current_fmt.fmt.pix_mp.pixelformat & 0xff,
-		    current_fmt.fmt.pix_mp.width, current_fmt.fmt.pix_mp.height, current_fmt.fmt.pix_mp.plane_fmt[0].bytesperline,
-		    current_fmt.fmt.pix_mp.field,
-		    current_fmt.fmt.pix_mp.plane_fmt[0].sizeimage);
-
-		stride = current_fmt.fmt.pix_mp.plane_fmt[0].bytesperline;
-	}
-
 
 	memset(&buf, 0, sizeof buf);
 	buf.type = (capture) ? V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE : V4L2_BUF_TYPE_VIDEO_OUTPUT_MPLANE;
@@ -730,12 +703,10 @@ vsp_queue_buffer(int fd, int capture, struct vsp_surface_state *vs)
 	buf.length = vs->base.num_planes;
 
 	memset(planes, 0, sizeof(planes));
-	for (i = 0; i < vs->base.num_planes; i++){
+	for (i = 0; i < vs->base.num_planes; i++) {
 		buf.m.planes[i].m.fd = vs->base.planes[i].dmafd;
-		if (!capture){
-			buf.m.planes[i].bytesused = stride * vs->base.height;
-			buf.m.planes[i].length = buf.m.planes[i].bytesused;
-		}
+		buf.m.planes[i].length = vs->base.planes[i].length;
+		buf.m.planes[i].bytesused = vs->base.planes[i].bytesused;
 	}
 
 	if (ioctl(fd, VIDIOC_QBUF, &buf) == -1) {
-- 
2.7.4

